src/material/schematics/ng-generate/navigation/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.html.template

```diff
@@ -5,9 +5,9 @@
       [opened]="(isHandset$ | async) === false">
     <mat-toolbar>Menu</mat-toolbar>
     <mat-nav-list>
-      <a mat-list-item href="#">Link 1</a>
-      <a mat-list-item href="#">Link 2</a>
-      <a mat-list-item href="#">Link 3</a>
+      <a mat-list-item routerLink="/">Link 1</a>
+      <a mat-list-item routerLink="/">Link 2</a>
+      <a mat-list-item routerLink="/">Link 3</a>
```

Comment:
I think we cannot do that as the schematic does not necessarily require `@angular/router` to be installed and set up.

Reply:
But the question is are we going to use href, the schematic will not throw an exception even if we make this change. And I replace it with routerLink every time

Reply:
I'm not sure. My thinking is that the schematic intends to provide a basic foundation for a navigation and that it's not obvious whether the Angular router is used or not.

I could imagine us having a schematic prompt for this though.

Reply:
We have have routerLink default and move to href based on input, i don't have stats, but i am sure most of the time, developers will use routerLink.

Reply:
I definitely agree that `routerLink` is commonly used for navigation, but my point is that `routerLink` does not work if `@angular/router` and the `RouterModule` are set up.

It might not fail having a `routerLink` attribute, but the `href` is not set if the router module is not set up. So the anchors will actually not be clickable. So ideally, we'd ensure `@angular/router` is set up, and setup the `RouterModule` if needed.

Reply:
I agree with @devversion that we can't assume that people will be using the router. If you go through `ng new`, you have to explicitly select that you want the router to be included.